### ISSUE TITLE
router.getMatchedComponents() doesn't seem to match alias

### ISSUE THREAD

**Issue by matthew-dean** [vuejs/vue-router#1488]
### Version
2.5.3

### Reproduction link
[https://codepen.io/matthewdean/pen/PjoeJw](https://codepen.io/matthewdean/pen/PjoeJw)

### Steps to reproduce
I have this router:
```
export function createRouter () {
  return new Router({
    mode: 'history',
    routes: [
      {
        path: '/', 
        component: Home,
        alias: '/index.html'
      },
      { 
        path: '/hello.html', 
        component: Hello
      }
    ]
  })
}
```
I'm using SSR to render Vue pages. However, when debugging, I've found that getMatchedComponents() only works with the main path, not the alias. So for this:
```
 // set server-side router's location
    router.push(context.url)

    // wait until router has resolved possible async components and hooks
    router.onReady(() => {
      const matchedComponents = router.getMatchedComponents()
```
... when `context.url` is '/' it renders my home component. When `context.url` is '/index.html` then matchedComponents is `[undefined]`.

### What is expected?
For getMatchedComponents() to match '/` and `/index.html`.

### What is actually happening?
It's not doing that.

**Comment by matthew-dean** [comment]
Note: the following works (some kind of internal logic bug?)
```js
export function createRouter () {
  return new Router({
    mode: 'history',
    routes: [
      {
        path: '/index.html', 
        component: Home,
        alias: '/'
      },
      { 
        path: '/hello.html', 
        component: Hello
      }
    ]
  })
}
```
So when '/' is the alias of '/index.html' it seems to work, rather than the reverse.

**Comment by posva** [comment]
You can provide a github repo as a repo, but make sure to boil it down, please 🙂

**Comment by LinusBorg** [comment]
Seems like related to / duplicate of #1427

**Comment by posva** [comment]
I thought so too, but it looks like he gets the opposite effect

**Comment by posva** [comment]
Closing due to inactivity. Please open a new issue with a reference to this one if you can follow up with more information.

**Comment by wilk** [comment]
This bug has been fixed in version 2.6.0: https://github.com/vuejs/vue-router/commit/8e30c34bb30e42984f8e7d1833c1a176794c1f0e
